frontend/src/pages/modelServing/screens/projects/useNIMAccountConstants.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { fetchNIMAccountConstants } from '~/pages/modelServing/screens/projects/nimUtils';
+
+type NIMAccountConstants =
+  | {
+      nimSecretName: string;
+      nimNGCSecretName: string;
+      nimConfigMapName: string;
+      templateName: string;
+    }
+  | undefined;
+
+export const useNIMAccountConstants = (dashboardNamespace: string): NIMAccountConstants => {
+  const [constants, setConstants] = React.useState<NIMAccountConstants>(undefined);
```

Comment:
Minor nitpick but I'm pretty sure you can remove the `undefined` in the `NIMAccountConstants` type and the useState parameter to `React.useState<NIMAccountConstants>()`  and typescript will be fine. the useState will implicitly have the ` ... | undefined`